### PR TITLE
Add support in Viewer for simple ContentState 1.0 constructs

### PR DESCRIPTION
**Pull request by mikeapp**
See https://gist.github.com/mikeapp/9a31e63a20c6f41e19a7f96b586f84cb for manual testing instructions.   I haven't tested the code beyond what's listed there.

### CODE REVIEW

**Comment by mathewjordan** [comment]
Thanks for submitting this @mikeapp!

It looks like you're still working. I tested it as-is and feels close.

**Comment by mathewjordan** [comment]
As another note, I have played around with this implementation in the past to parse `iiifContent`. I'm not sure if this is helpful, but it could abstract much of the logic to `iiif.ts`

```
const parseIiifContent = (iiifContent: string) => {
  let resourceId;
  let active;
  if (isURL(iiifContent)) {
    resourceId = iiifContent;
    active = {
      manifest: resourceId,
      canvas: undefined,
    };
  } else if (isEncoded(iiifContent)) {
    const json = JSON.parse(decodeContentState(iiifContent));
    switch (json?.type) {
      // https://iiif.io/api/content-state/1.0/#51-a-region-of-a-canvas-in-a-manifest
      // https://iiif.io/api/content-state/1.0/#52-start-playing-at-a-point-in-a-recording
      // https://iiif.io/api/content-state/1.0/#53-multiple-targets-for-a-comparison-view
      // https://iiif.io/api/content-state/1.0/#54-search-results
      case "SpecificResource":
      case "Range":
      case "Annotation":
        resourceId = json?.target.partOf[0].id;
        active = {
          manifest: resourceId,
          canvas: json?.id,
        };
        break;
      case "Canvas":
        resourceId = json?.partOf[0].id;
        active = {
          manifest: resourceId,
          canvas: json?.id,
        };
        break;
      case "Manifest":
        resourceId = json?.id;
        break;
    }
  }
  return { resourceId, active };
};
```

**Comment by mikeapp** [comment]
Thanks for the code example.   I'll take a look as I have time later this week.

**Comment by mikeapp** [comment]
@mathewjordan  I think this is ready for review.

File: src/components/Viewer/index.tsx [hunk 1]
```diff
@@ -157,7 +163,8 @@ const RenderViewer: React.FC<CloverViewerProps> = ({
       if (manifests.length > 0) {
         dispatch({
           type: "updateActiveManifest",
-          manifestId: manifests[0],
+          manifestId:
+            decodeContentStateManifestURI(iiifContent) || manifests[0],
```
Comment (mathewjordan) [inline src/components/Viewer/index.tsx:167]: Likewise with Canvas, Clover has issues of the Manifest doesn't exist on the Collection. I wonder if we can work this into a single helper: 

```suggestion
  getActiveManifest(iiifContent, manifests),
```

and then in `iiif.ts`

```ts
export const getActiveManifest = (iiifContent: string, manifests: string[]) => {
  const manifest = decodeContentStateManifestURI(iiifContent);
  return manifests.includes(manifest) ? manifest : manifests[0];
};
```

File: src/components/Viewer/index.tsx [hunk 2]
```diff
@@ -109,7 +114,8 @@ const RenderViewer: React.FC<CloverViewerProps> = ({
           setManifest(data);
           dispatch({
             type: "updateActiveCanvas",
-            canvasId: data.items[0] && data.items[0].id,
+            canvasId:
+              decodeContentStateCanvasURI(iiifContent) || data.items[0]?.id,
```
[the remark this answers is not in the document]
Comment (mathewjordan) [reply src/components/Viewer/index.tsx:118]: Following the example below, perhaps we also add a getActiveCanvas convenience function here: 

```tsx
  const canvases = data.items.map((item) => item.id);
  dispatch({
    type: "updateActiveCanvas",
    canvasId: getActiveCanvas(iiifContent, canvases),
  });
```

```ts
export const getActiveCanvas = (iiifContent: string, canvases: string[]) => {
  const canvas = decodeContentStateCanvasURI(iiifContent);
  return canvases.includes(canvas) ? canvas : items[0];
};
```

**Review by mathewjordan — CHANGES_REQUESTED** [review]
@mikeapp This is excellent. Thank you so much for implementing this change.

Note my comments about determining Canvas and Manifest exist as partOf their respective Manifest or Collection. Perhaps this could happen in the helpers I noted.

Also noted:

- Line 130, the useEffect dependency list needs `iiifContent` and should read `[activeManifest, iiifContent, dispatch, vault]`.
- Line 176, the useEffect dependency list needs `iiifContent` and should read `[dispatch, iiifContent, iiifResource]`.

**Review by mathewjordan — APPROVED** [review]

**Comment by mathewjordan** [comment]
@mikeapp This is excellent. I'm going to roll this up with some basic documentation and release as v2.5.0. Thank you for the contribution.

**Pull request merged**